lib/beautydate_api/api_session.rb

```diff
@@ -0,0 +1,45 @@
+module BeautydateApi
+  class APISession
+
+    def authenticate(authorization, email, password)
+      request = {
+        method: 'POST',
+        url: "#{BeautydateApi.base_uri}sessions",
+        timeout: 30,
+        headers: {
+          user_agent: "BeautyDate/#{BeautydateApi::VERSION}; Ruby Client",
+          content_type: 'application/vnd.api+json',
+          authorization: authorization
+        },
+        payload: {
+          data: {
+            type: 'sessions',
+            attributes: {
+              provider: 'b2beauty',
+              email: email,
+              password: password
+            }
+          }
+        }.to_json
+      }
+      result = JSON.parse(RestClient::Request.execute request)
+      @token_key = result['data']['attributes']['token']
+      @expires_at = result['data']['attributes']['expires_at']
+      true
+    rescue => e
+      raise AuthenticationException
+    end
+
+    def authenticated?
+      !@token_key.to_s.empty?
+    end
+
+    def valid?
+      authenticated? and Time.now <= Time.at(@expires_at)
```

Comment:
Sugestão pra gente não fugir ao [style guide](https://github.com/bbatsov/ruby-style-guide#no-and-or-or):

```ruby
authenticated? && Time.at(@expires_at) > Time.now
```